### ISSUE TITLE
[1.12.2 Modded] Moving Motion Controllers Too Quickly Lags/Crashes Client

### ISSUE THREAD

**Issue by AwesomeAnimationGamer** [jrbudda/Vivecraft_112#77]
I am running a modded server on 1.12.2 with Vivecraft support. Ingame everything runs quite smoothly - unless I'm moving my Touch controllers too quickly. Swinging my arm for things like attacking a mob results in massive FPS stutter, and it's quite common for a common action like roomscale mining to cause the game to freeze completely and then crash. The crash log (attached below) seems to indicate that Vivecraft itself is the culprit. If you could check this over and let me know what the source of this crash is, and how I can get it resolved?

Thanks in advance for your assistance.
[crash-2018-12-03_23.35.25-client.txt](https://github.com/jrbudda/Vivecraft_112/files/2642460/crash-2018-12-03_23.35.25-client.txt)

**Comment by AwesomeAnimationGamer** [comment]
Do you have any ideas as to the cause of this crash? I'm getting tired of crashing just because I was roomscale mining...

**Comment by jrbudda** [comment]
well its a LittleTiles left click event encountering a null object. I will have to try to reproduce it. Can you determine what mods are needed for it to crash? just littletiles?

**Comment by AwesomeAnimationGamer** [comment]
Oh I see... So basically I should make a modpack with just Vivecraft and LittleTiles and just roomscale mine and see if I crash?

**Comment by AwesomeAnimationGamer** [comment]
I just created a modpack with an identical Forge version to the larger modpack, and installed only LittleTiles (and CreativeCore, since it's a required dependency). It only took around 30 seconds for the game to crash once I started running around waving my arms and punching things.
[crash-2018-12-09_00.57.18-client.txt](https://github.com/jrbudda/Vivecraft_112/files/2660279/crash-2018-12-09_00.57.18-client.txt)

**Comment by jrbudda** [comment]
that would be helpful

**Comment by AwesomeAnimationGamer** [comment]
If it helps any further, the Forge version used on my server is 14.23.5.2793

**Comment by jrbudda** [comment]
Yes so this is a problem. Basically when Forge generates a LeftClick event for mods to use, it calculates its own raytrace based on your position and look direction to tell the mod the exact vector that was hit.  I guess most mods don't use it, but littletiles needs it of course to determine subtiles. The problem is the Forge raytrace can return null, and LittleTiles is not checking for a null. 

When you physically mine the hit vector has nothing to do with your look direction, nor the crosshair direction, it's the trajectory of your tool, which of course forge has no way to know about. So the raytrace it generates is always garbage, and sometimes null, and it crashes in the null case. 

I'm not sure how to fix this from our end, but I'll think about it.

**Comment by jrbudda** [comment]
[vivecraft-1.12.2-jrbudda-7-r8-installer.zip](https://github.com/jrbudda/Vivecraft_112/files/2666182/vivecraft-1.12.2-jrbudda-7-r8-installer.zip)
Id be interested to know if this solves the problem without introducing any others.

**Comment by AwesomeAnimationGamer** [comment]
I have installed it to the same test instance I used to confirm the crash. I will test it this evening after work. Thanks!
